contributors/design-proposals/mount-options.md

```diff
@@ -35,14 +57,37 @@ Mount options can be specified as `mount-options` annotations in PV. For example
       server: 172.17.0.2
 ```
 
+### Mount option support in Storage Classes
+
+Kubernetes admin can also specify mount option as a parameter in storage class.
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: slow
+provisioner: kubernetes.io/glusterfs
+parameters:
+  type: gp2
+  mountOptions: "auto_mount"
```

Comment:
I think `mountOptions` should be a first class field in the `StorageClass` because it is controlling a first class field in the PV object (not specific to a single volume plugin). As far as validation is concerned, for in-tree plugins you can call `SupportsMountOption(...)` on the provisioner. The tricky bit will be how to pass this through for out-of-tree volume provisioners, but that problem needs to be solved for reclaimPolicy (https://github.com/kubernetes/kubernetes/pull/47987) as well.